src/game_api/spawn_api.cpp

```diff
@@ -662,18 +662,38 @@ void init_spawn_hooks()
     }
 }
 
-void spawn_player(int8_t player_slot, float x, float y)
+int32_t spawn_player(int8_t player_slot, std::optional<float> x, std::optional<float> y, std::optional<LAYER> layer)
 {
     if (player_slot < 1 || player_slot > 4)
-        return;
+        return -1;
+    auto state = State::get().ptr();
+    auto& slot = state->items->player_select_slots[player_slot - 1];
+    if (slot.character < to_id("ENT_TYPE_CHAR_ANA_SPELUNKY") || slot.character > to_id("ENT_TYPE_CHAR_CLASSIC_GUY"))
+        return -1;
+    if (state->items->player_count < player_slot)
+        state->items->player_count = player_slot;
+    slot.activated = true;
 
     push_spawn_type_flags(SPAWN_TYPE_SCRIPT);
     OnScopeExit pop{[]
                     { pop_spawn_type_flags(SPAWN_TYPE_SCRIPT); }};
 
-    using spawn_player_fun = void(Items*, uint8_t ps, float pos_x, float pos_y);
+    auto old_x = state->level_gen->spawn_x;
+    auto old_y = state->level_gen->spawn_y;
+    state->level_gen->spawn_x = x.value_or(old_x);
+    state->level_gen->spawn_y = y.value_or(old_y);
+    auto uid = (int32_t)state->next_entity_uid;
```

Comment:
HH are spawned first, if you really need the uid returned you can just get it from `state.items.players`

Reply:
It just seemed weird that it's the only spawn function that doesn't return uid and after learning all the things it really spawns, still sounds logical to return the player uid.